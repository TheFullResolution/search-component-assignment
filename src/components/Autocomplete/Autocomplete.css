.autocomplete-container {
  position: relative;
  width: 100%;
}

.autocomplete-wrapper {
	border: 1px solid rgb(210, 210, 210);
  border-top: none;
  max-height: 300px;
	overflow-y: auto;
  position: absolute;
  width: 100%;
}

.autocomplete-item {
  border-bottom: 1px solid rgb(210, 210, 210);
  padding: 8px;
}

.autocomplete-highlighted {
  background-color: rgba(210, 210, 210, 0.5);
}

.autocomplete-searchterm {
  font-weight: 700;
}

.autocomplete-highlight {
  background-color: inherit;
  color: rgb(113, 113, 113);
  font-weight: 400;
}

.autocomplete-counter {
  color: rgb(204, 204, 204);
}
