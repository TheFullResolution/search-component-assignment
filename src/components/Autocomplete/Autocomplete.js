// @flow
import './Autocomplete.css'

import React from 'react'
import Highlighter from 'react-highlight-words'
import type { AutoCompleteItem } from '../Value/Value'
import { TransitionGroup } from 'react-transition-group'
import { DropDownAnimation } from '../Animations/DropDownAnimation'
import { FadeAnimation } from '../Animations/FadeAnimation'

type Props = {
  autoComplete: AutoCompleteItem[],
  getItemProps: (params: any) => any,
  highlightedIndex: number,
  show: boolean,
  inputValue: string
}

export const Autocomplete = ({
  autoComplete,
  getItemProps,
  highlightedIndex,
  inputValue,
  show
}: Props) => (
  <div className="autocomplete-container">
    <TransitionGroup>
      {show ? (
        <DropDownAnimation>
          <div className="autocomplete-wrapper">
            <TransitionGroup>
              {autoComplete.map((item, index) => (
                <FadeAnimation key={item.searchterm}>
                  <div
                    {...getItemProps({ item: item.searchterm })}
                    className={`autocomplete-item ${
                      highlightedIndex === index
                        ? 'autocomplete-highlighted'
                        : ''
                    }`}
                  >
                    <Highlighter
                      highlightClassName="autocomplete-highlight"
                      searchWords={[inputValue]}
                      className="autocomplete-searchterm"
                      autoEscape={true}
                      textToHighlight={item.searchterm}
                    />
                    <span> ({item.nrResults})</span>
                  </div>
                </FadeAnimation>
              ))}
            </TransitionGroup>
          </div>
        </DropDownAnimation>
      ) : null}
    </TransitionGroup>
  </div>
)
