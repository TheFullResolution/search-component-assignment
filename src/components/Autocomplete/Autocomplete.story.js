// @flow

import React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import { linkTo } from '@storybook/addon-links'
import {Autocomplete} from './Autocomplete.js'

storiesOf('Autocomplete', module).add('default', () => <Autocomplete />)
