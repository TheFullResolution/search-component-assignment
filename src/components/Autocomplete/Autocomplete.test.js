// @flow

import React from 'react'
import renderer from 'react-test-renderer'
import { Autocomplete } from './Autocomplete.js'

const props = {
  autoComplete: [
    { searchterm: 'test', nrResults: 10 },
    { searchterm: 'try', nrResults: 10 },
    { searchterm: 'another', nrResults: 10 }
  ],
  getItemProps: params => params,
  highlightedIndex: 0,
  show: true,
  inputValue: 'test'
}

it('Autocomplete: default', () => {
  const component = renderer.create(<Autocomplete {...props} />)
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
