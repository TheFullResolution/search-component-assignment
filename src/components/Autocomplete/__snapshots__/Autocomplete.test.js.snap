// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete: default 1`] = `
<div
  className="autocomplete-container"
>
  <div>
    <div
      className="autocomplete-wrapper"
    >
      <div>
        <div
          className="autocomplete-item autocomplete-highlighted"
          item="test"
        >
          <span
            className="autocomplete-searchterm"
          >
            <mark
              className="autocomplete-highlight "
              style={Object {}}
            >
              test
            </mark>
          </span>
          <span>
             (
            10
            )
          </span>
        </div>
        <div
          className="autocomplete-item "
          item="try"
        >
          <span
            className="autocomplete-searchterm"
          >
            <span
              className=""
              style={undefined}
            >
              try
            </span>
          </span>
          <span>
             (
            10
            )
          </span>
        </div>
        <div
          className="autocomplete-item "
          item="another"
        >
          <span
            className="autocomplete-searchterm"
          >
            <span
              className=""
              style={undefined}
            >
              another
            </span>
          </span>
          <span>
             (
            10
            )
          </span>
        </div>
      </div>
    </div>
  </div>
</div>
`;
